py2many/rewriters.py

```diff
@@ -138,7 +138,7 @@ def visit_If(self, node):
             if hasattr(node, "scopes") and len(node.scopes) > 1:
                 rename(node.scopes[-2], "main", "main_func")
             # ast.parse produces a Module object that needs to be destructured
-            ret = ast.parse("def main(): True").body[0]
+            ret = ast.parse("def main(argc: int, argv: List[str]): True").body[0]
```

Comment:
Is pycpp the only one which uses argc? If so, we can probably simplify a bit by removing argc here and handling it inside pycpp,

Reply:
Not it isnt.

Reply:
```
$ grep argc tests/expected/sys_argv.* -l
tests/expected/sys_argv.cpp
```

which other backend is using argc?

sys_argv* tests pass for me even after removing `argc` in the line above. What breaks if we remove it from all the py*/transpiler.py also in this diff?

Reply:
Ah, backend, yes, none are using it currently.

However many languages have the ability to implement the C `main` directly.
e.g. nim has `proc cmain(c: cint, a: cstringArray): cint {.importc.}`.

My rust voodoo avoided me needing to do that, but we probably will want to switch to it for Rust to avoid the `Box::leak`.

Creating a new signature for `main` doesnt make sense.  There is already one global definition of the `main` entry point: the C one, which exists in every program on nearly every OS out there.

Reply:
Every language has it's own signature for main. Most of them (6 out of 7) different from C and they have their own way to map/interop with C main if necessary. Rust has concepts like: `#[start]` that may map to what you're thinking.

https://github.com/rust-lang/rust/pull/5809

When we transpile, we map python's main signature to the target language's main signature, which is what your code is doing.

This isn't a show stopper for me. Since you have other stuff on top of this, let's get this in and work on simplifying the code in later diffs.